### ISSUE TITLE
Questions about source code

### ISSUE THREAD

**Issue by soloice** [XMUNLP/XMUNMT#4]
I've read through a large part of the source code, and find it to be well-organized and easy to read.
But I still have a few questions. Could you please verify them?

1, Is the following comment correct?
 https://github.com/XMUNLP/XMUNMT/blob/ed7ae9aff721769246072ef8e5457d34eb82e5d5/xmunmt/layers/attention.py#L76
As far as I understand, the shape of `hidden` in L74 is [batch_size * mem_size, hidden_size], `logits` in L77 is of shape [batch_size * mem_size, 1], `logits` in L78 is of shape [batch_size, mem_size], and no variable has a shape of [batch_size, mem_size, 1]. Or am I getting it wrong?

2, In https://github.com/XMUNLP/XMUNMT/blob/ed7ae9aff721769246072ef8e5457d34eb82e5d5/xmunmt/models/rnnsearch.py#L280, should it be incremental decoding? In this case, the decoder just run for one step.

**Comment by Playinf** [comment]
1. You're correct, the shape of logits is indeed [batch_size, mem_size]. That comment is misleading and I'll fix it in the next commit.
2. The current version does not support incremental decoding. As a result, the beam search function will call model_graph at every step and recompute source annotation and previous hidden states. However, the decoding speed is acceptable. Using non-incremental decoder will make the implementation of RNNsearch easier (only need to write one graph). The special case for non-incremental decoding saves the computation of softmax (only use the last position).

**Comment by soloice** [comment]
Oh, I see. It's just an API limitation. You wanted to perform incremental decoding during inference phase, but current interfaces don't support this feature. This can be worked around by re-designing APIs.
Thanks~
